evm/chains/base.py

```diff
@@ -71,23 +88,27 @@ class BaseChain(Configurable, metaclass=ABCMeta):
     """
     The base class for all Chain objects
     """
+    chaindb = None  # type: BaseChainDB
+
     #
     # Chain Initialization API
     #
     @classmethod
     @abstractmethod
     def from_genesis(cls,
-                     chaindb,
-                     genesis_params,
-                     genesis_state=None):
+                     chaindb: BaseChainDB,
+                     genesis_params: Dict,
+                     genesis_state: Dict[Address, Dict]=None) -> 'BaseChain':
```

Comment:
Can we specify the type of the inner Dict or if not make it explicitly `Dict[Any, Any] `. *Implicit* Any is not allowed once we migrate to stricter checks (for good reasons)